### PR TITLE
[TESTS] Added a test case to ensure special characters can be used du…

### PR DESCRIPTION
**Pull request by markkellyeire**
…ring signature creation LRN-36469

## Checklist

- [ ] Feature
- [ ] Bug
- [ ] Security
- [ ] Documentation

- [ ] ChangeLog.md updated

- [x ] Tests added
- [x] All testsuites passed
- [ ] `make dist` completed successfully

### CODE REVIEW

File: tests/integration/test_dataapi.py [hunk 1]
```diff
@@ -18,7 +18,9 @@
 items_request = {
     # These items should already exist for the demos consumer
     'references': ['item_2', 'item_3'],
-    'limit': 1
+    'limit': 1,
+    # Add a name that does not contain special characters
+    'name': 'Tect'
```
Comment (shtrom) [inline tests/integration/test_dataapi.py:23]: `name` is not a valid parameter for `GET items` https://reference.learnosity.com/data-api/endpoints/itembank_endpoints#getItems, so there's no guarantee that 
1. the signature spans that (it does, for now, but not sure in the future),
2. api-data would accept it without error (ditto).

I'd suggest not doing anything here, and `appending` a special-charactered reference to `items_request['references']` in `test_real_request_with_special_characters`.

File: tests/integration/test_dataapi.py [hunk 2]
```diff
@@ -78,6 +80,23 @@ def test_paging(self):
         assert len(results) == 2
         assert results == {'item_2', 'item_3'}
 
+    def test_real_request_with_special_characters(self):
+        """Make a request against Data Api to ensure the SDK works"""
+        client = DataApi()
+
+        # Change the name to one containing special characters to ensure
+        # signature creation works with special characters in the request
+        items_request['name'] = 'тест'
```
Comment (shtrom) [inline tests/integration/test_dataapi.py:89]:
```suggestion
        local_items_request = items_request.copy()  # prevent modifying the base fixture
        local_items_request['references'] = items_request['references'].copy()  # prevent modifying the base fixture's list
        local_items_request['references'].append('тест')
```

File: tests/integration/test_dataapi.py [hunk 3]
```diff
@@ -78,6 +80,23 @@ def test_paging(self):
...
+
+        res = client.request(self.__build_base_url() + items_endpoint, security, consumer_secret, items_request,
+                             action)
+        returned_json = res.json()
+
+        assert len(returned_json['data']) > 0
```
Comment (shtrom) [inline tests/integration/test_dataapi.py:95]: I think this test code is starting to be duplicated a fair bit. Soon might be the time to DRY it. Not for now, though.

**Review by shtrom — CHANGES_REQUESTED** [review]

File: tests/integration/test_dataapi.py (hunk 1 above)
Comment (markkellyeire) [reply tests/integration/test_dataapi.py:23]: Good shout.

File: tests/integration/test_dataapi.py (hunk 3 above)
Comment (markkellyeire) [reply tests/integration/test_dataapi.py:95]: Agreed. I will create a ticket for Groot as any familiarity with the SDK's is valuable at this stage.

File: tests/integration/test_dataapi.py (hunk 2 above)
Comment (markkellyeire) [reply tests/integration/test_dataapi.py:89]: Much better, I will go with that.

**Review by shtrom — APPROVED** [review]

**Pull request merged**